### DISCUSSION TITLE
Qual o assunto mais difícil no front-end?

### DISCUSSION THREAD

**Discussion by pauloviniciusdev** [frontendbr/forum#2122] [Discussão]
Olá! Me chamo Paulo, e gostaria de perguntar a você, participante deste fórum, qual o assunto/tecnologia/tarefa/linguagem mais difícil no front end. ATENÇÃO: não precisa ter uma resposta correta e absoluta, isso é apenas uma pergunta em que a resposta deve ser apenas a sua opinião. Eai? qual a matéria que teve mais dificuldade?

**Comment by nazarepiedady** [comment]
Saber trabalhar e bem com CSS, isso quero dizer otimizar coisas que você normalmente faria com JavaScript (que satura a projeto rapidamente)!
Posso afirmar que 80% dos profissionais da área e isso incluindo os que vêm se formando, não sabem escrever folhas de estilos coerentemente, aplicar suas potencialidades e por isso vivem com eterno ódio (meio adolescente) da tecnologia, tanto que hoje existem muletas para esses deficientes, como CSS-In-JS, já que é o que mais gostam e falam, o que não adianta muito, porque se não se sabe escrever uma boa marcação (HTML) e estilos para essa marcação (CSS), o seu JavaScript será uma droga ou mesmo um absurdo assim como rodinhas para bicicletas de adultos.

Estude muito bem, HTML, CSS e JavaScript 😉.

**Reply by pauloviniciusdev** [reply]
Valeu ai, amigo. 17 dias e ninguém me respondeu, apenas você... Obrigado ai. Pelo menos consegui uma resposta. Já tava pensando se tinha feito algo de errado... tipo... sei lá, um palavrão talvez... Obrigado ai pela humildade.

**Reply by nazarepiedady** [reply]
> Valeu ai, amigo. 17 dias e ninguém me respondeu, apenas você... Obrigado ai. Pelo menos consegui uma resposta. Já tava pensando se tinha feito algo de errado... tipo... sei lá, um palavrão talvez... Obrigado ai pela humildade.

@pauloviniciusdev busco sempre que possível ajudar quem está começando, e caso estiver precisando de algum mentor que lhe acompanhe, eu estou disponível pelo meu Facebook [nazarepiedady](https://www.facebook.com/nazarepiedady).

**Reply by Juuhz** [reply]
Boa @nazarepiedady !

**Comment by Juuhz** [comment]
Vou botar fogo no parquinho descordando do colega @nazarepiedady haha ❤️ .

Saber trabalhar bem com CSS é de fato algo de suma importância para o ecossistema front-end, vejo muito valor, mas também enxergo isso como algo básico, uma skill que você vai aprimorando ao passar do tempo, já ta no sangue do Front.

Hoje, a web de maneira geral, está muito mais focada em performance/conteúdo, consequência do aumento drástico que tivemos com os avanços das tecnologias móveis.

E esse é o tópico do norral front-end que eu acho mais desafiador: **Como entregar uma página performática**.

No mercado atualmente, temos vários frameworks que nos ajudam a abstrair algumas boas práticas, oque de fato, ajuda muito, mas não resolve todo o pepino.

Quando falamos de um BLOG, é bem fácil imaginar esse caminho de "arquitetura ideal", mas e quando falamos de uma plataforma ou algo mais robusto? Começa a complicar né?

Um bom front, tem que ter uma noção básica de arquitetura, para poder entregar páginas cada vez mais performáticas.

- Quando renderizar uma página no servidor?
- Quando utilizar geração de página stática ou sob demanda? (SSG do NextJs, por exemplo).
- Quando utilizar Cache ( CDNs, SWR, etc... ).
- Interfaces otimistas.
- Otimização de imagens, css, fontes, etc...
- Componentização de códigos.
- Lazy loading / Skeleton.
- Middlewares / BFF ( Best For Front-end ).

Essas perguntas podem ser um bom ponto de partida, para estudos.

Por fim, um bom Front vai sempre entregar um bom CSS, é natural, mais oque vai fazer ele se destacar, é o carinho que ele vai ter, para sempre entregar algo de valor e performático para o usuário final, e nem sempre otimizações de css conseguem alcançar esse resultado.

**Reply by pauloviniciusdev** [reply]
Brabo... Eu estou no começo dos meus estudos em front-end e já estava achando que o front-end não era tão profundo assim.

**Reply by Juuhz** [reply]
Rapaz, te falar, é bem profundo!! haha.

Quando começamos a falar de Autenticação, revalidação de token, segurança, dados, dashboard, etc... A parada começa a ficar + difícil.

**Reply by Juuhz** [reply]
Mas agora que você ta no começo, é uma boa focar no básico mesmo, feijão com arroz ( HTML, CSS, JS ), dps tu vai avançando para as Libs / Frameworks (e de preferência, escolha 1 pra tu focar). 

Com o tempo, as experiências que você for vivendo no mercado de trabalho, vão te fazer feras nesses tópicos que pontuei. 

Ah, e nunca deixe de consumir conteúdo, tecnologia é uma eterna mutação!

**Reply by pauloviniciusdev** [reply]
Mano, valeu ai. Uma coisa muito top das comunidades de programação que percebi é que vocês costumam ajudar muito, principalmente os iniciantes.
